test/parallel/test-domain-timers-uncaught-exception.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that the timer callbacks are
+// called in the order in which they were created
+// in the event of an unhandled exception in the domain
+
+const domain = require('domain').create();
+const assert = require('assert');
+
+var first = false;
+
+domain.run(function() {
+  setTimeout(function() { throw Error('FAIL'); }, 1);
+  setTimeout(function() { first = true; }, 1);
+  setTimeout(function() { assert.strictEqual(first, true);  }, 2);
```

Comment:
do es6 functions break the test case? you are using a mixture of es6 and normal js, there is a wave of es6-ization going though the test suite ATM, better catch it and make these es6, or comment on why they must be function